src/classes/thumbnail.py

```diff
@@ -136,19 +151,19 @@ def do_GET(self):
 
         # Locate thumbnail
         thumb_path = os.path.join(info.THUMBNAIL_PATH, file_id, "%s.png" % file_frame)
-        if not os.path.exists(thumb_path) and file_frame == 1:
-            # Try with no frame # (for backwards compatibility)
-            thumb_path = os.path.join(info.THUMBNAIL_PATH, "%s.png" % file_id)
-        if not os.path.exists(thumb_path) and file_frame != 1:
-            # Try with no frame # (for backwards compatibility)
-            thumb_path = os.path.join(info.THUMBNAIL_PATH, "%s-%s.png" % (file_id, file_frame))
+        if not os.path.exists(thumb_path):
+            if file_frame == 1:
+                # Try ID with no frame # (for backwards compatibility)
+                alt_path = os.path.join(info.THUMBNAIL_PATH, "%s.png" % file_id)
+            else:
+                # Try with ID and frame # in filename (for backwards compatibility)
+                alt_path = os.path.join(info.THUMBNAIL_PATH, "%s-%s.png" % (file_id, file_frame))
+
+            if os.path.exists(alt_path):
+                thumb_path = alt_path
```

Comment:
@jonoomph 

This relates to this item: "Fix output path for thumbnail generation (don't override with legacy/compatibility path)"

I noticed that, because `thumb_path` was being overridden by the legacy locations **before** it was passed to `GenerateThumbnail()`, the thumbnails were always being generated at the "backwards compatibility" path. That's now fixed, and thumbnails should be created as `ProjectName_assets/thumbnail/FILE_ID/frame#.png` as presumably intended.